packages/aws-cdk/lib/context-providers/security-groups.ts

```diff
@@ -12,18 +12,45 @@ export class SecurityGroupContextProviderPlugin implements ContextProviderPlugin
     const account: string = args.account!;
     const region: string = args.region!;
 
+    if (args.securityGroupId && args.securityGroupName) {
+      throw new Error('\'securityGroupId\' and \'securityGroupName\' can not be specified both when looking up a security group');
+    }
+
+    if (!args.securityGroupId && !args.securityGroupName) {
+      throw new Error('\'securityGroupId\' or \'securityGroupName\' must be specified to look up a security group');
+    }
```

Comment:
Should we remove or keep those checks? In module `ec2` I removed them because no public API allows to pass both or none of these parameters.

I would propose to keep the checks in this module. If somebody implements a custom construct using the logic from the context-providers, it would be helpful if these checks are available.

Reply:
Agreed; worth keeping here down at this level.